acceptance/tests/face/4654_facts_face.rb

```diff
@@ -0,0 +1,57 @@
+test_name "Puppet facts face should resolve custom and external facts"
+
+#
+# This test is intended to ensure that custom and external facts present
+# on the agent are resolved and displayed by the puppet facts face.
+#
+custom_fact = <<CFACT
+Facter.add('custom_fact') do
+  setcode do
+    'foo'
+  end
+end
+CFACT
+
+unix_external_fact = <<EFACT
+#!/bin/sh
+echo 'external_fact=bar'
+EFACT
+
+win_external_fact = <<EFACT
+@echo off
+echo external_fact=bar
+EFACT
+
+agents.each do |agent|
+  if agent['platform'] =~ /windows/
+    external_fact = win_external_fact
+    ext = '.bat'
+  else
+    external_fact = unix_external_fact
+    ext = '.sh'
+  end
+
+  step "Create custom and external facts in their default directories on the agent"
+  on agent, puppet('apply'), :stdin => <<MANIFEST
+  file { "#{agent.puppet['plugindest']}/facter":
+    ensure => directory,
+  }
+
+  file { "#{agent.puppet['plugindest']}/facter/custom.rb":
```

Comment:
@er0ck Updated. Verified that we start out with no `facter` dir in `plugindest`, and that `pluginfactdest` exists before the test, hence blowing away only the external fact.